CORE-5408 Simplify CpiIdentifier endpoint types

barrylapthorn opened the pull request:
Reduce CpiIdentifier endpoint types to 1.

Introduce new extension function to convert packaging CpiIdentifier
to (JSON serializable) endpoint type that's user friendly.

Remove duplicates.

barrylapthorn committed:
CORE-5408 Simplify CpiIdentifier endpoint types

Reduce CpiIdentifier endpoint types to 1.

Introduce new extension function to convert packaging CpiIdentifier
to (JSON serializable) endpoint type that's user friendly.

Remove duplicates.

barrylapthorn commented on processors/rpc-processor/src/integrationTest/resources/swaggerBaseline.json:
```diff
@@ -2677,16 +2637,16 @@
         "description" : "Details of the virtual node to be created"
       },
       "CreateVirtualNodeResponse" : {
```
Since this is in the diff @vkolomeyko  - should this be opened as a Jira issue?  There are several `CreateXYZ` entries in here which seem oddly named for a REST api.

vkolomeyko replied:
I do not find it particularly odd... @barrylapthorn - how would you rather have `CreateVirtualNodeResponse` named?

barrylapthorn replied:
Well, it's the first time I've seen it and having a _data object_ prefixed with an _action verb_ is definitely odd.

Just feels like we need to remove the prefix.

vkolomeyko replied:
Possibly, this method could just return `VirtualNodeInfo` ? (then we will not need `CreateVirtualNodeResponse` type at all)

barrylapthorn replied:
They're not a  1:1  mapping but I think using `VirtualNodeInfo` (endpoint type) is possibly fine - the only values we can't return are `version` and `timestamp` in the code _as is_.

Both come from this:

```
    @HttpRpcPOST(
        title = "Create virtual node",
        description = "Creates a new virtual node.",
        responseDescription = "The details of the created virtual node."
    )
    fun createVirtualNode(
        @HttpRpcRequestBodyParameter(description = "Details of the virtual node to be created")
        request: CreateVirtualNodeParameters
    ): CreateVirtualNodeResponse
```
    
I think they (the `Create`) objects should just be `VirtualNodeRequest` and `VirtualNodeResponse`, or like you said `VirtualNodeInfo`

Outside of the scope of this PR but worth raising a jira issue?

vkolomeyko replied:
Yes, probably makes sense to raise a Jira on that. Thank you!

barrylapthorn replied:
https://r3-cev.atlassian.net/browse/CORE-5886

vkolomeyko commented on libs/virtual-node/cpi-upload-endpoints/build.gradle:
```diff
@@ -11,6 +11,7 @@ dependencies {
     api project(':libs:http-rpc:http-rpc')
 
     implementation platform("net.corda:corda-api:$cordaApiVersion")
+    api project(':libs:packaging:packaging-core')
```
Not immediately obvious to me from GitHub diff... but why do we need to put this as `api` dependency?

barrylapthorn replied:
It's a transient dependency so that its version of `CpiIdentifier` is exposed in the vnode endpoints project.

vkolomeyko replied:
@barrylapthorn - I have created a PR #1710 targeting your branch to illustrate my point.

barrylapthorn replied:
Good catch.  I didn't like the `packaging-core` part, so that's gone now.

vkolomeyko commented on libs/virtual-node/virtual-node-endpoints/build.gradle:
```diff
@@ -10,6 +10,7 @@ dependencies {
 
     api project(':libs:http-rpc:http-rpc')
 
+    api project(":libs:virtual-node:cpi-upload-endpoints")
```
Not immediately obvious to me from GitHub diff... but why do we need to put this as `api` dependency?

barrylapthorn replied:
It's also transient dependency so that the (endpoint) version of `CpiIdentifier` is exposed in the vnode endpoints project.  I'll double check and change back to `implementation` though.

barrylapthorn replied:
Yes, if I change it back to `implementation` I get a compilation error - 

```
e: /home/barry/dev/r3/corda-runtime-os/components/virtual-node/virtual-node-rpcops-service-impl/src/main/kotlin/net/corda/virtualnode/rpcops/impl/v1/VirtualNodeRPCOpsImpl.kt: (107, 17): Cannot access class 'net.corda.libs.cpiupload.endpoints.v1.CpiIdentifier'. Check your module classpath for missing or conflicting dependencies
```

vkolomeyko commented on libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/types/VirtualNodeInfo.kt:
```diff
@@ -0,0 +1,33 @@
+    val timestamp: Instant,
+)
+
+fun net.corda.virtualnode.VirtualNodeInfo.toEndpointType() : VirtualNodeInfo =
```
Would it not be a better idea to move both `toEndpointType()` into `impl` module where it is currently used?

Commit:
CORE-5408: Improvements to the modules packaging (#1710)

In particular, dependency: `api project(':libs:packaging:packaging-core')` is a problem as intention is to keep HTTP endpoints as lean as possible in terms of dependencies.

This might be important, if Java HTTP RPC Client is created against the interface it will not need to pull an extensive graph of transitive dependencies.

vkolomeyko reviewed (approved):
LGTM
